cmd/cinder_flexvolume_driver

```diff
@@ -0,0 +1,308 @@
+#!/bin/bash
+
+# Copyright 2020 Authors of Arktos.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Notes:
+# This is simple flex volume driver to demostrate use of Cinder static volume
+# for Arktos cluster along with Arktos vm runtime
+# It uses Cinder client and Cinder extended APIs for local attach/detach support
+# package required:
+# JQ
+# python-cinderclient
+# python-brick-cinderclient-ext
+#
+usage() {
+        err "Invalid usage. Usage: "
+        err "\t$0 init"
+        err "\t$0 attach <json params> <nodename>"
+        err "\t$0 detach <mount device> <nodename>"
+        err "\t$0 waitforattach <mount device> <json params>"
+        err "\t$0 mountdevice <mount dir> <mount device> <json params>"
+        err "\t$0 unmountdevice <mount dir>"
+        err "\t$0 isattached <json params> <nodename>"
+        exit 1
+}
+
+err() {
+        echo -ne $* 1>&2
+}
+
+log() {
+        echo -ne $* >&1
+}
+
+logfile() {
+	echo $(date -u +%Y-%m-%dT%H:%M:%SZ) "INFO: " $*  >> /tmp/cinderflexvol_driver.log
+}
+
+#
+# Set the environment variables per your test env
+# Those are used by the cinder client
+#
+setEnv() {
+	export OS_PROJECT_ID=3ab2fbaaf99d422f980c97c4b5092f4b
+    	export OS_USER_DOMAIN_ID=default
+    	export OS_AUTH_URL=http://172.31.14.52/identity
+    	export OS_PASSWORD=Futurewei@123
+    	export OS_USERNAME=admin
+}
+
+getDevice() {
+	VOLUMEID=$1
+	setEnv
+	DMDEV=$(/usr/local/bin/cinder get-volume-paths ${VOLUMEID})
+	echo $(getRealDevice ${DMDEV})
+}
+
+getRealDevice() {
+	readlink -f $1
+}
+
+attach() {
+        logfile "Attach cinder volume to local host" "Args $*"
+        JSON_PARAMS=$1
+        logfile "JSON_PARAMS: " $JSON_PARAMS
+
+	VOLUMEID=$(echo ${JSON_PARAMS} | jq -r '.cinderVolumeID')
+	logfile "Got volumeID: "  ${VOLUMEID}
+
+	VOLUMEPATH=$(echo ${JSON_PARAMS} | jq -r '.path')
+        logfile "Got volumePath: "  ${VOLUMEPATH}
```

Comment:
How does the cinder client attach the volume to "${VOLUMEPATH}"? I didn't see the variable is used in the function attach().

Reply:
will remove. it was added to experiment link to the device so that we won't need to manipulate the path filed when generating the virtlet-flexvolume.json file. it is not used anymore. will remove.